src/main/java/tigase/xmpp/impl/SaslAuth2.java

```diff
@@ -484,7 +489,7 @@ private BareJID extractUserJid(final Exception e, XMPPResourceConnection session
 
 	private void saveIntoBruteForceLocker(final XMPPResourceConnection session, final Exception e) {
 		try {
-			if (bruteForceLocker.isEnabled(session)) {
+			if (bruteForceLocker != null && bruteForceLocker.isEnabled(session)) {
```

Comment:
As you have more than one place you are doing the same check (but in different files), I would think about adding method `bool isBruteForceLockerEnabled(XMPPResourceConnection)` and storing `null` check and `isEnabled()` inside this method.